crates/ruff_python_formatter/tests/snapshots/format@expression__binary_implicit_string.py.snap

```diff
@@ -0,0 +1,284 @@
+        b,
+        c,
+        d,
+    )
+)
+(
+    "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+    "bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb"
+    "cccccccccccccccccccccccccccccccccccccccccccc"
+    % aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
+)
+
+
+def test():
+    return (
+        "\n%(modified_count)s %(identifier)s %(action)s"
+        "%(destination)s%(unmodified)s%(post_processed)s."
+    ) % {
+        "modified_count": modified_count,
+        "identifier": "static file" + ("" if modified_count == 1 else "s"),
+        "action": "symlinked" if self.symlink else "copied",
+        "destination": (" to '%s'" % destination_path if destination_path else ""),
+        "unmodified": (
+            ", %s unmodified" % unmodified_count if collected["unmodified"] else ""
+        ),
+        "post_processed": (
+            collected["post_processed"] and ", %s post-processed" % post_processed_count
+            or ""
+        ),
+    }
+
+
+# trailing expression comment
+self._assert_skipping(
+    SkipTestCase("test_foo").test_foo,
+    ValueError,
+    "skipUnlessDBFeature cannot be used on test_foo (test_utils.tests."
+    "SkippingTestCase.test_skip_unless_db_feature.<locals>.SkipTestCase%s) "
+    "as SkippingTestCase.test_skip_unless_db_feature.<locals>.SkipTestCase "
+    "doesn't allow queries against the 'default' database."
+    # Python 3.11 uses fully qualified test name in the output.
+    % (".test_foo" if PY311 else ""),
+)
+
+# dangling operator comment
+self._assert_skipping(
+    SkipTestCase("test_foo").test_foo,
+    ValueError,
+    "skipUnlessDBFeature cannot be used on test_foo (test_utils.tests."
+    "SkippingTestCase.test_skip_unless_db_feature.<locals>.SkipTestCase%s) "
+    "as SkippingTestCase.test_skip_unless_db_feature.<locals>.SkipTestCase "
+    "doesn't allow queries against the 'default' database."
+    %  # Python 3.11 uses fully qualified test name in the output.
+    (".test_foo" if PY311 else ""),
+)
+
+# Black keeps as many operands as fit on the same line as the `%`. Ruff does not. This is intentional as these are rare and complicated things significantly
+(
+    "aaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+    "bbbbbbbbbbbbbbbbbbbbbbbbbbbbb"
+    "cccccccccccccccccccccccccc"
+    % aaaaaaaaaaaa
+    + x
+)
+
+(
+    b
+    + c
+    + d
+    + "aaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+    "bbbbbbbbbbbbbbbbbbbbbbbbbbbbb"
+    "cccccccccccccccccccccccccc" % aaaaaaaaaaaa
+    + x
+)
+
+(
+    b
+    < c
+    > d
+    < "aaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+    "bbbbbbbbbbbbbbbbbbbbbbbbbbbbb"
+    "cccccccccccccccccccccccccc" % aaaaaaaaaaaa
+    > x
+)
+
+
+self.assertEqual(
+    response.status_code,
+    status_code,
+    msg_prefix
+    + "Couldn't retrieve content: Response code was %d"
+    " (expected %d)" % (response.status_code, status_code),
+)
+
+
+def test():
+    return (
+        (
```

Comment:
These parentheses are incorrect but they get inserted by the tuple formatting. This is probably similar to the *nested binary expression* case in that the string formatting should only enforce the optional parentheses of the `return` statement but not from the tuples.